bgpd/bgpd.c

```diff
@@ -1996,6 +1999,17 @@ int peer_remote_as(struct bgp *bgp, union sockunion *su, const char *conf_if,
 	return 0;
 }
 
+const char *get_name_by_role(uint8_t role)
+{
+	static const char *const bgp_role_names[] = {
+		"provider", "rs-server", "rs-client", "customer", "peer"};
+	if (role == ROLE_UNDEFINE)
+		return "undefine";
+	if (role <= 5)
+		return bgp_role_names[role];
+	return "unknown";
```

Comment:
What is the difference between unknown and undefined?

Reply:
An undefined role is a situation where no role has been assigned at all.

If someone creates a new draft with new roles and send a Capability with that new role number, that role will be considered as "Unknown". So, Unknown role was created for future forward compatibility.